pkg/api/handlers/compat/info.go

```diff
@@ -133,6 +134,37 @@ func GetInfo(w http.ResponseWriter, r *http.Request) {
 	utils.WriteResponse(w, http.StatusOK, info)
 }
 
+func getServiceConfig(runtime *libpod.Runtime) *registry.ServiceConfig {
+	var indexConfs map[string]*registry.IndexInfo
+
+	regs, err := sysregistriesv2.GetRegistries(runtime.SystemContext())
+	if err == nil {
+		indexConfs = make(map[string]*registry.IndexInfo, len(regs))
+		for _, reg := range regs {
+			mirrors := make([]string, len(reg.Mirrors))
+			for i, mirror := range reg.Mirrors {
+				mirrors[i] = mirror.Location
+			}
+			indexConfs[reg.Prefix] = &registry.IndexInfo{
+				Name:    reg.Prefix,
+				Mirrors: mirrors,
+				Secure:  !reg.Insecure,
+			}
+		}
+	} else {
+		log.Warnf("failed to get registries configuration: %v", err)
+		indexConfs = make(map[string]*registry.IndexInfo)
+	}
+
+	return &registry.ServiceConfig{
+		AllowNondistributableArtifactsCIDRs:     make([]*registry.NetIPNet, 0),
+		AllowNondistributableArtifactsHostnames: make([]string, 0),
+		InsecureRegistryCIDRs:                   make([]*registry.NetIPNet, 0),
+		IndexConfigs:                            indexConfs,
+		Mirrors:                                 make([]string, 0),
```

Comment:
why did not improve this, my daemon.json
```
 cat /etc/docker/daemon.json 
{
  "graph": "/data/kube/docker",
  "registry-mirrors": ["https://registry.docker-cn.com", "https://docker.mirrors.ustc.edu.cn"],
  "insecure-registries":["reg.xxxx.lan:5000", "xxx.yun.xxx.cn"],
  "log-driver": "json-file",
  "log-level": "warn",
  "bip": "10.185.0.1/16",
  "log-opts": {
    "max-size": "10m",
    "max-file": "3",
    "tag": "{{.ImageName}}|{{.Name}}|{{.ImageFullID}}|{{.FullID}}"
  }
}
```
```
{
  "AllowNondistributableArtifactsCIDRs": [],
  "AllowNondistributableArtifactsHostnames": [],
  "InsecureRegistryCIDRs": [
    "127.0.0.0/8"
  ],
  "IndexConfigs": {
    "docker.io": {
      "Name": "docker.io",
      "Mirrors": [
        "https://registry.docker-cn.com/",
        "https://docker.mirrors.ustc.edu.cn/"
      ],
      "Secure": true,
      "Official": true
    },
    "reg.xxx.lan:5000": {
      "Name": "reg.xxx.lan:5000",
      "Mirrors": [],
      "Secure": false,
      "Official": false
    },
    "xxx.yun.xxx.cn": {
      "Name": "xxx.yun.xxx.cn",
      "Mirrors": [],
      "Secure": false,
      "Official": false
    }
  },
  "Mirrors": [
    "https://registry.docker-cn.com/",
    "https://docker.mirrors.ustc.edu.cn/"
  ]
}
```

Reply:
I am interested only in "IndexConfigs".